lib/pavilion/test_config/resolver.py

```diff
@@ -440,6 +449,106 @@ def load_raw_configs(self, tests, host, modes):
+
+                    if (FULL_RE.match(highest) is not None or
+                        ABBRV_RE.match(highest) is not None):
+                        highest = [int(i) for i in highest.split(".")]
+                    else:
+                        err=("'{}.{}' has invalid value: "
+                             "\n'{}' in compatible_pav_versions '{}' "
+                             "is not a valid version format."
+                             .format(test_suite, test, highest,
+                                     compatible_versions))
+                        raise TestConfigError(err)
+
+                    if not (lowest <= pav_version <= highest):
+                        compatible_errors += err.format(test,
+                                                        PavVars.version(self),
+                                                        compatible_versions)
+
+                # Version Provided with wildcard. EX: 2.1.*
+                elif '.*' in compatible_versions:
+                    if WC_RE.match(compatible_versions) is not None:
+                        version = [int(i) for i in compatible_versions.split(".")[:-1]]
+                        pav_version = pav_version[:-1]
+                    else:
+                        err=("'{}.{}' has invalid value: "
+                             "\ncompatible_pav_versions '{}' in not a "
+                             "valid version format."
+                             .format(test_suite, test, compatible_versions))
+                        raise TestConfigError(err)
+
+                    if not pav_version == version:
+                        compatible_errors += err.format(test,
+                                                        PavVars.version(self),
+                                                        compatible_versions)
+
+                # Specific Version Provided. EX: 2.1.3, 2.1
+                else:
+                    if FULL_RE.match(compatible_versions) is not None:
+                        version = [int(i) for i in compatible_versions.split(".")]
+                    elif ABBRV_RE.match(compatible_versions) is not None:
+                        version = [int(i) for i in
+                                   compatible_versions.split(".")[:-1]]
+                        pav_version = pav_version[:-1]
+                    else:
+                        err=("'{}' in '{}' has invalid value: "
+                              "\ncompatible_pav_versions '{}' is not a "
+                              "valid version format."
+                             .format(test, test_suite, compatible_versions))
+                        raise TestConfigError(err)
+
+                    if not pav_version == version:
+                        compatible_errors += err.format(test,
+                                                        PavVars.version(self),
+                                                        compatible_versions)
+
```

Comment:
Regardless of the multiple regular expressions, this should still be a lot simpler. 
 - Figure out a discrete set of steps to follow. 
 - The output from each step should be normalized to minimize special cases in the next step.
 - Break a step out into a function only if it's complicated enough that you can't tell what the steps' code does at
   a glance, if the function will be re-used, or to further divide the problem into sub-steps. 

1. get_min_str('1.2.*-3.4') -> '1.2.*'
2. get_max_str('1.2.*-3.4') -> '3.4'
3. calc_min(min_str) -> (1, 2, 0)
4. calc_max(max_str) -> (3, 4, None)
5. check_version(version, min_version, max_version) -> True|False

Reply:
Draw table is both a good and bad example. It's good in that it follows discrete steps to get to the end result. It's bad in that each of the larger steps should be broken out into separate functions, which is mostly a result of the organic way in which the function grew over time.